tests/jballerina-unit-test/src/test/resources/test-src/functions/check_in_parameter_default_negative.bal

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+isolated function intOrError() returns int|error => error("err!");
+
+function f1(int a = check intOrError()) returns int => a;
```

Comment:
In the case of the following type of code, 
```ballerina
isolated function intOrError() returns int => 1;

function f1(int a = check intOrError()) returns int => a;
```

Only the error, `invalid usage of the 'check' expression operator: no expression type is equivalent to error type`, is thrown. Is it expected to throw the error introduced in this PR in this case also

Reply:
It's a warning, but yeah, we give only that even for records also. IIRC that was because they won't fail at runtime, but for full compliance and to avoid surprises (if `intOrError` changes to return an error, it will result in a compilation error for the use of `check`, which is kinda unrelated to the change), will disallow in the non-error case also.

Reply:
Changed in [9fe1874](https://github.com/ballerina-platform/ballerina-lang/pull/43185/commits/9fe18743e878f0ac083505b28479c0418a6bdec8).